src/gui/game/Brush.h

```diff
@@ -57,45 +59,38 @@ class Brush
 	virtual void SetRadius(ui::Point radius)
 	{
 		this->radius = radius;
-		this->size = radius+radius+ui::Point(1, 1);
+		this->size = radius + radius + ui::Point(1, 1);
 
 		GenerateBitmap();
 		updateOutline();
 	}
-	virtual ~Brush() {
-		delete[] bitmap;
-		delete[] outline;
-	}
+	virtual ~Brush() = default;
 	virtual void RenderRect(Renderer * ren, ui::Point position1, ui::Point position2);
 	virtual void RenderLine(Renderer * ren, ui::Point position1, ui::Point position2);
 	virtual void RenderPoint(Renderer * ren, ui::Point position);
 	virtual void RenderFill(Renderer * ren, ui::Point position);
 	virtual void GenerateBitmap()
 	{
-		delete[] bitmap;
-		bitmap = new unsigned char[size.X*size.Y];
-		for(int x = 0; x < size.X; x++)
-		{
-			for(int y = 0; y < size.Y; y++)
-			{
-				bitmap[(y*size.X)+x] = 255;
-			}
-		}
+		bitmap = std::vector<unsigned char>(size.X * size.Y);
+		std::fill_n(std::back_inserter<std::vector<unsigned char>>(bitmap), size.X * size.Y, 255);
```

Comment:
The size is known ahead of time, why not use `vector<T>::vector(size_t, T)`?

Reply:
@mniip even better, why not array<T, size_t>